Library/Homebrew/language/java.rb

```diff
@@ -1,8 +1,16 @@
 module Language
   module Java
-    def self.java_home_env(version=nil)
+    def self.java_home_cmd(version=nil)
       version_flag = " --version #{version}" if version
-      { :JAVA_HOME => "$(/usr/libexec/java_home#{version_flag})" }
+      "/usr/libexec/java_home#{version_flag}"
+    end
+
+    def self.java_home_env(version=nil)
+      { :JAVA_HOME => "$(#{java_home_cmd(version)})" }
+    end
+
+    def self.overridable_java_home_env(version=nil)
+      { :JAVA_HOME => "$(if [ -z \"$JAVA_HOME\" ]; then echo \"$(#{java_home_cmd(version)})\"; else echo \"$JAVA_HOME\"; fi)" }
```

Comment:
Thanks @bfontaine  

@mikemcquaid The script is definitely using bash. c.f. https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/extend/pathname.rb#L376-L377